### ISSUE TITLE
CVANAL paths

### ISSUE THREAD

**Issue by tjingboem** [csound/csound#1493]
According to the manual page on Utilities https://csound.com/manual/UtilityTop.html:

_Resultant analysis files are written into the current directory, or to the named directory if a path is included._

CVANAL is the only utility i found that does not comply to this rule; i have to give full pathname or look in SADIR for the analyzed file. The analyzed file is not written to the path the example is running from.
It is looking, but the analyzed file is not there:

```
cannot load rv_stereo1.con
INIT ERROR in instr 2 (opcode convolve.i) line 37: CONVOLVE cannot load rv_stereo1.con
 from file /media/menno/datae/onderzoek/manualExamples/CVANAL.csd (1)
```
Here is the file:
[rv_stereo.wav.zip](https://github.com/csound/csound/files/6685902/rv_stereo.wav.zip)

This is the csd:

```
<CsoundSynthesizer>
<CsOptions>
; Select audio/midi flags here according to platform
-odac   -m0  --limiter=.99 ;;;realtime audio out, with limiter protection
; For Non-realtime ouput leave only the line below:
; -o cvanal.wav -W ;;; for file output any platform
</CsOptions>
<CsInstruments>

sr = 44100
ksmps = 32
nchnls = 2
0dbfs  = 1

; by Menno Knevel 2021

gilen  filelen "rv_stereo.wav"	    ; get length of impulse soundfile

; analyze sound file and output result to 3 convolve files
ires1 system_i 1,{{ cvanal rv_stereo.wav rv_stereo1.con }}                  ; default settings
ires2 system_i 1,{{ cvanal -d.15 rv_stereo.wav rv_stereo2.con }}            ; use only first portion
ires3 system_i 1,{{ cvanal -b.25 -d.001 rv_stereo.wav rv_stereo3.con }}    ; take very short portion from the end


instr 1 ; untreated signal
asig    diskin2   "beats.wav", 1
prints  "\n---***YOU NOW HEAR THE UNTREATED SOUND SAMPLE***---\n"
outs    asig, asig
endin

instr 2

prints  "\n---***YOU NOW HEAR THE RESULT OF THIS ANALYZED FILE:***---\n"
prints "--**used duration of impulse file = %5.3f seconds (total =%5.2f)**--\\n", p5, gilen

adry    diskin2 "beats.wav"                 ; input (dry) audio
awet1, awet2 convolve adry*.8, p4           ; stereo convolved (wet) audio
awet1  *=  p6                               ; scale amplitude of impulse sound, Left
awet2  *=  p6                               ; & Right channel
adrydel delay   adry, p5                    ; delay dry signal to align it with convolved signal
outs    (adrydel+awet1)*.8,(adrydel+awet2)*.8 ; mix wet & dry signals 
endin

</CsInstruments>
<CsScore>

i1 1 2          ; untreated signal

i2 4 3  "rv_stereo1.con"    0.39    .01	; use total impulse
i2 8 3  "rv_stereo2.con"    0.15    .01	; first portion of impulse
i2 12 3 "rv_stereo3.con"    0.001    1	; very short portion of impulse, starting from 0.25 sec, & scale amp
e
</CsScore>
</CsoundSynthesizer>
```

All of the output:

```
0dBFS level = 32768.0
--Csound version 6.16 (double samples) Jun  6 2021
[commit: none]
libsndfile-1.0.28
UnifiedCSD:  /media/menno/datae/onderzoek/manualExamples/CVANAL.csd
STARTING FILE
Creating options
Creating orchestra
closing tag
Creating score
rtaudio: ALSA module enabled
rtmidi: ALSA Raw MIDI module enabled
sorting score ...
	... done
graphics suppressed, ascii substituted
0dBFS level = 1.0
0dBFS level = 32768.0
--Csound version 6.16 (double samples) Jun  6 2021
[commit: none]
libsndfile-1.0.28
util cvanal:
audio sr = 44100, stereo, reading both channels
opening WAV infile rv_stereo.wav
analysing 17302 sample frames (0.4 secs)
cvanal finished
end of score.		   overall amps:      0.0
	   overall samples out of range:        0
0 errors in performance
Elapsed time at end of performance: real: 0.006s, CPU: 0.006s
0dBFS level = 32768.0
--Csound version 6.16 (double samples) Jun  6 2021
[commit: none]
libsndfile-1.0.28
util cvanal:
audio sr = 44100, stereo, reading both channels
opening WAV infile rv_stereo.wav
analysing 6615 sample frames (0.1 secs)
cvanal finished
end of score.		   overall amps:      0.0
	   overall samples out of range:        0
0 errors in performance
Elapsed time at end of performance: real: 0.002s, CPU: 0.002s
0dBFS level = 32768.0
--Csound version 6.16 (double samples) Jun  6 2021
[commit: none]
libsndfile-1.0.28
util cvanal:
audio sr = 44100, stereo, reading both channels
opening WAV infile rv_stereo.wav
analysing 44 sample frames (0.0 secs) from timepoint 0.2
cvanal finished
end of score.		   overall amps:      0.0
	   overall samples out of range:        0
0 errors in performance
Elapsed time at end of performance: real: 0.000s, CPU: 0.000s
orch now loaded
audio buffered in 256 sample-frame blocks
writing 512 sample blks of 64-bit floats to dac
SECTION 1:

---***YOU NOW HEAR THE UNTREATED SOUND SAMPLE***---

---***YOU NOW HEAR THE RESULT OF THIS ANALYZED FILE:***---
--**used duration of impulse file = 0.390 seconds (total = 0.39)**--
cannot load rv_stereo1.con
INIT ERROR in instr 2 (opcode convolve.i) line 37: CONVOLVE cannot load rv_stereo1.con
 from file /media/menno/datae/onderzoek/manualExamples/CVANAL.csd (1)
awet1	awet2	convolve.i	#a0	p4	0	
	  B  4.000 - note deleted.  i2 had 1 init errors

---***YOU NOW HEAR THE RESULT OF THIS ANALYZED FILE:***---
--**used duration of impulse file = 0.150 seconds (total = 0.39)**--
cannot load rv_stereo2.con
INIT ERROR in instr 2 (opcode convolve.i) line 37: CONVOLVE cannot load rv_stereo2.con
 from file /media/menno/datae/onderzoek/manualExamples/CVANAL.csd (1)
awet1	awet2	convolve.i	#a0	p4	0	
	  B  8.000 - note deleted.  i2 had 1 init errors

---***YOU NOW HEAR THE RESULT OF THIS ANALYZED FILE:***---
--**used duration of impulse file = 0.001 seconds (total = 0.39)**--
cannot load rv_stereo3.con
INIT ERROR in instr 2 (opcode convolve.i) line 37: CONVOLVE cannot load rv_stereo3.con
 from file /media/menno/datae/onderzoek/manualExamples/CVANAL.csd (1)
awet1	awet2	convolve.i	#a0	p4	0	
	  B 12.000 - note deleted.  i2 had 1 init errors
Score finished in csoundPerform().
inactive allocs returned to freespace
end of score.		   overall amps:  0.99997  0.99997
	   overall samples out of range:        0        0
3 errors in performance
2068 512 sample blks of 64-bit floats written to dac
```

**Comment by tjingboem** [comment]
Quotes from include Csound Analyze Utilities in examples #541:

Menno:
I have been thinking about this some more, and in https://csound.com/manual/OrchDirFiles.html it is established that:
* _Soundfiles for reading are sought in the current directory. Or SSDIR or SFDIR_
So they should be written to one of these places so they can be found.
* _Analysis control files for reading are sought in the current directory. Or SADIR_
So they should be written to one of these place in order to be found.

For backward compatibilty reasons this should not change, imo.

All the utily examples i made (ATSA, HETRO, LPANAL, PVANAL) write their Analysis file to the current dir.
So, when running these examples from the current dir, all analyzed files are found.

CVANAL should write his analyzed file to current dir or SADIR. But it is not. instead It writes to SFDIR. 
So, when running the example it checks not the current dir nor SFDIR.

(Now i must say that on my system i have had issues for Csound to find files from SFDIR, SSDIR and SADIR for a long time. 
So it may be that this fault is on my system only. But still CVANAL is not writing to the right dirs)

As CVANAL does not write to current dir, as opposed to the other Utilities, it can not be found when trying to look for it. 
And the Analyzed file is not written to SADIR but to SFDIR instead- wrong dir, and it can not be found:
```
cannot load rv_stereo13.con
INIT ERROR in instr 2 (opcode convolve.i) line 37: CONVOLVE cannot load rv_stereo13.con
 from file /media/menno/datae/onderzoek/manualExamples/cvanalJOHN.csd (1)
```

These are my env:
SSDIR=/media/menno/datae/csounddata/Samples
SADIR=/media/menno/datae/csounddata/Analyses
SFDIR=/media/menno/datae/csounddata/Output


Is the example working out of the box on your machine? Where does it write the Analyzed file to?



John:
Are you suggesting that cvanal should write to SADIR (rather than SFDIR) or curent directory?



Menno:
I am suggesting that cvanal should be writing to the place where all utilities write to.
I don't know why the behavior of cvanal is different in that respect. At least, on my system.

All other analyze utilities on my computer write to the current directory; the directory where the example is running from.
This provides clean examples that run out-of-the-box on all systems for every user.
cvanal is the exception, and i wondered why and if that is correct.

In short: cvanal should be writing to current directory. AND be looked for using lpread etc.
